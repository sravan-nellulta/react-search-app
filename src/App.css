.header-container {
  background-color:#0000002e;
  height: 80px
}
.header{
  text-align: left;
}

.company-name {
  margin: 0px;
}

.company-motto {
  margin: 0px;
}

.header-date {
  position: absolute;
  right: 0px;
  top:40px;
}

.employee-heading {
  display: inline-block;
  width: 50%;
}

.inline-block {
  float: left;
  width: 50%;
}

.image-block {
  width: 150px;
}

.employee-sort {
  display: inline-block
}

.select-box {
  margin-left: 20px;
  margin-right: 20px;
  width: 101px;
}
.employee-data {
  width: 32%;
  float: left;
  border: 1px solid gray;
  margin: 5px;
  height: 300px
  /* padding: 25px; */
}
.employee-data:hover {
  background-color: #EEE;
  cursor: pointer
}

/* .bio {
  width:340px
} */

.close {
  position: absolute;
  top:0;
  right: 0;
  cursor: pointer;
}

.modal-img {
  padding: 25px;
}
.modal-details {
  display: inline-block;
  margin: 15px;
}
.modal-jb-title {
  margin:0;
}

.modal-data {
  display: inline-block;
  top:50px;
  position: absolute;
}

.modal {
  position: fixed;
  top: 0;
  left: 0;
  width:100%;
  height: 100%;
  background: rgba(0, 0, 0, 0.6);
  display:block;
}
.modal-main {
  position:fixed;
  background: white;
  width: 80%;
  height: auto;
  top:50%;
  left:50%;
  transform: translate(-50%,-50%);
}