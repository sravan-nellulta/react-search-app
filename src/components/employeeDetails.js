import React from 'react';


const employeeDetails = (props) => {
    const { data: { firstName, avatar, bio, lastName }, onClick } = props;
    return (
        <div className='employee-data' onClick={onClick}>
            <div className='inline-block'>
                <img src={avatar} alt={firstName} className='image-block' />
            </div>
            <div className='inline-block bio'>
                <p>{firstName} {lastName}</p>
                <p>{bio}</p>
            </div>
        </div>)
}


export default employeeDetails;
