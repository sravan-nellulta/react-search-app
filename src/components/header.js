import React from 'react';
import * as JsonData from '../assets/sample-data.json';

const header = () => {
  return (
    <div className='header-container'>
      <div className='header'>
        <h3 className='company-name'>{JsonData.companyInfo.companyName}</h3>
        <p className='company-motto'>{JsonData.companyInfo.companyMotto}</p>
      </div>
      <p className='header-date'>Since: {JsonData.companyInfo.companyEst}</p>
    </div>)
}

export default header;
