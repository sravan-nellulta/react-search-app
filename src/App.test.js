import React from 'react';
// import ReactDOM from 'react-dom';
import {shallow} from 'enzyme';
import App from './App';
import header from './components/header';
import employeeDetails from './components/employeeDetails';

const Enzyme = require('enzyme');
const EnzymeAdapter = require('enzyme-adapter-react-15');

// Setup enzyme's react adapter
Enzyme.configure({ adapter: new EnzymeAdapter() });

describe('header', () => {
  /* it('should render app component', () => {
    const appComponent = shallow(
      <App className='App' />
      );
      expect(appComponent).toMatchSnapshot();
  }); */
  it('should be defined', () => {
    expect(employeeDetails).toBeDefined();
  });
  it('should render correctly', () => {
    const tree = shallow(
      <employeeDetails className='employee-data' />
    );
    expect(tree).toMatchSnapshot();
  });
  it('should be defined', () => {
    expect(header).toBeDefined();
  });
  it('should render correctly', () => {
    const tree = shallow(
      <header className='header-container' />
    );
    expect(tree).toMatchSnapshot();
  });
 });
