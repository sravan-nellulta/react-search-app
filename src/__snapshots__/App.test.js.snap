// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`header should render correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <header
    className="header-container"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "className": "header-container",
    },
    "ref": null,
    "rendered": null,
    "type": "header",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "className": "header-container",
      },
      "ref": null,
      "rendered": null,
      "type": "header",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactFifteenAdapter {
      "options": Object {
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "prevContext": true,
          },
        },
        "supportPrevContextArgumentOfComponentDidUpdate": true,
      },
    },
  },
}
`;

exports[`header should render correctly 2`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <header
    className="header-container"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "className": "header-container",
    },
    "ref": null,
    "rendered": null,
    "type": "header",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "className": "header-container",
      },
      "ref": null,
      "rendered": null,
      "type": "header",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactFifteenAdapter {
      "options": Object {
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "prevContext": true,
          },
        },
        "supportPrevContextArgumentOfComponentDidUpdate": true,
      },
    },
  },
}
`;
